File: oracle/app/entrypoints/api/endpoints/metrics/health.ts
import { Router } from "express";
import { HealthStatusResponse } from "../../schemas/health";

export function registerHealthEndpoint(router: Router): void {
  router.get("/metrics/health", (_req, res) => {
    const payload: HealthStatusResponse = {
      status: "healthy",
      timestamp: new Date().toISOString()
    };

    res.json(payload);
  });
}

File: prediction_engine/app/entrypoints/api/endpoints/metrics/health.py
"""API endpoint for health checks."""

from datetime import datetime, timezone

from fastapi import APIRouter

from app.entrypoints.api.schemas.health import HealthStatusResponse

router = APIRouter(tags=["Metrics"])


@router.get(
    "",
    summary="API health check.",
    response_model=HealthStatusResponse,
)
async def health_check() -> HealthStatusResponse:
    """Returns the health status of the API."""

    return HealthStatusResponse(
        status="healthy", timestamp=datetime.now(timezone.utc).isoformat()
    )
